### ISSUE TITLE
Getting webpack 5 errors with new ERB project. Did I miss something?

### ISSUE THREAD

**Issue by chrisle** [electron-react-boilerplate/electron-react-boilerplate#3055]
## Summary

When starting a brand new project with ERB and attempting to use the ipcRenderer, I would get the following error. What I'm not sure is if something is missing in the boilerplate or was there some missed setup I had not done.

```
ERROR in ./node_modules/electron/index.js 1:11-24
Module not found: Error: Can't resolve 'fs' in 'C:\Users\chris\code\test-project\node_modules\electron'
 @ ./src/renderer/App.tsx 12:24-43
 @ ./src/renderer/index.tsx 10:30-46

ERROR in ./node_modules/electron/index.js 2:13-28
```

1. Run `git clone --depth 1 --branch main https://github.com/electron-react-boilerplate/electron-react-boilerplate.git test-project`
2. `cd test-project`
3. `yarn`
4. Modify `src/renderer/App.tsx` as follows:

```ts
import { MemoryRouter as Router, Switch, Route } from 'react-router-dom';
import icon from '../../assets/icon.svg';
import './App.css';

import { ipcRenderer } from 'electron'; // Add this.

// Add this function
function sendTest() {
  console.log('test');
  ipcRenderer.send('asynchronous-message', 'test');
}

const Hello = () => {
  return (
    <div>
      <button onClick={sendTest}>Send Test</button> {/* Add this button */}
```

Reading other posts suggests editing the webpack.config.js file but one doesn't exist. Where does it go and what do I need to add to it?

**Comment by kbrisso** [comment]
Hi @chrisle 

The IPC render is not something documented clearly when using Typescript and takes some research to figure out. I spent a lot of time on this too. In order to use the IPC in App.tsx  or any tsx file, you need to add a global types file with interfaces along with a global Window object declaration.  

Please take a look at this [file ](https://github.com/kbrisso/erb-bootstrap-ipc/blob/main/src/renderer/global.d.ts) in my example project. You can see the Global Windows interface and how I declared the types. The interface signature matches the signature of the preload.js file. 

This is the link to the project itself
https://github.com/kbrisso/erb-bootstrap-ipc

In closing if you plan on using IPC a lot I came across this IPC method Invoke. Much cleaner to use, I hope this helps.



`// Renderer process`
`ipcRenderer.invoke('some-name', someArgument).then((result) => {`
`// ...`
`})`
`// Main process`
`ipcMain.handle('some-name', async (event, someArgument) => {`
`const result = await doSomeWork(someArgument)`
`return result`
`})`

**Comment by dclipca** [comment]
Alternatively https://www.electronjs.org/docs/latest/tutorial/context-isolation#usage-with-typescript

**Comment by kbrisso** [comment]
@amilajack 

What about adding an IPC render type file and a sample ping message in the App.jsx file? Coming from a non Electron world like me I found this a little challenging myself and I think it might be useful.  If you ever need help with updating the help docs let me know.  

@dclipca 

Thanks for the link. I found that too after the fact, coming from a non Electron world and Typescript experience made searching for what my issues was limited.
